src/Microsoft.Identity.Client/Internal/MsalHelpers.cs

```diff
@@ -186,6 +187,15 @@ public static string ToQueryParameter(this IDictionary<string, string> input)
             return builder.ToString();
         }
 
+        public static Dictionary<string, string> FromQueryParameter(string uri)
```

Comment:
@rayluo Regex is hard to read. The code already exists. I don't quite understand what you mean by "feel the same". The suggested method does EXACTLY what you need. Please do not introduce new functions just because "it feels different". Provide a clear case where the suggested function does not work for you. @addev-ashish - FYI